src/stories/select/Select.tsx

```diff
@@ -83,6 +67,55 @@ export type SelectProps<T = Record<string, any>> = {
     }
 } & Omit<InputProps, 'type' | 'overrides'>;
 
+export type SingleSelectProps<T = Record<string, any>> = {
+    /**
+     * The option ID to render as selected in the select box.
+     *
+     * This should exactly match the option IDs passed in the `options` prop. If `null`, no option will be selected.
+     */
+    value?: Option<T>['id'] | null;
+    /**
+     * The interaction handler for the Select.
+     */
+    onChange?: (value: Option<T>['id'] | null) => void | Promise<void>;
+    /**
+     * The visual variant of the Select. `listbox` will render as a dropdown menu, `radio` will render as a radio group, `card` will render as selectable cards, and `segmented` will render as a segmented control.
+     * @default listbox
+     */
+    kind?: 'listbox' | 'radio' | 'card' | 'segmented';
+    multiple?: false;
+    multipleItemsName?: never;
+} & CommonSelectProps;
+
+export type MultiSelectProps<T = Record<string, any>> = {
+    /**
+     * Controls the text of the Multiselect button when multiple items selected.
```

Comment:
/nit but I think would be good to clarify that this either gets prepended by `All` or `${value.length} ${multiItems}`. But wondering also if we should allow a React element to be passed - say if we wanted to do pills or something